lib/llm/src/http/service/openapi_docs.rs

```diff
@@ -0,0 +1,528 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! OpenAPI documentation generation and Swagger UI integration
+//!
+//! This module provides automatic OpenAPI specification generation from the HTTP service routes
+//! and serves Swagger UI for interactive API documentation.
+//!
+//! ## Features
+//!
+//! - **OpenAPI 3.0 Specification**: Automatically generates OpenAPI spec from defined routes
+//! - **Swagger UI**: Interactive API documentation accessible via web browser
+//! - **Dynamic Route Documentation**: Introspects registered routes and generates documentation
+//!
+//! ## Endpoints
+//!
+//! The module exposes two main endpoints:
+//!
+//! - `GET /openapi.json` - Returns the OpenAPI specification in JSON format
+//! - `GET /docs` - Serves the Swagger UI interface for interactive API exploration
+//!
+//! ## Configuration
+//!
+//! The OpenAPI documentation endpoints use fixed paths:
+//! - `/openapi.json` - The OpenAPI specification
+//! - `/docs` - The Swagger UI documentation interface
+//!
+//! ## Example Usage
+//!
+//! The OpenAPI documentation is automatically integrated into the HTTP service.
+//! Once the service is running, you can:
+//!
+//! 1. View the raw OpenAPI spec: `curl http://localhost:8000/openapi.json`
+//! 2. Access Swagger UI: Open `http://localhost:8000/docs` in a web browser
+
+use axum::Router;
+use utoipa::OpenApi;
+use utoipa::openapi::{PathItem, Paths, RefOr};
+
+use crate::http::service::RouteDoc;
+
+/// OpenAPI documentation structure
+///
+/// This struct defines the complete OpenAPI specification for the Dynamo HTTP service.
+/// It includes all the schemas, paths, and metadata needed to document the API.
+#[derive(OpenApi)]
+#[openapi(
+    info(
+        title = "NVIDIA Dynamo OpenAI Frontend",
+        version = "0.6.0",
```

Comment:
It might not work in this context, but ideally this would be `env!("CARGO_PKG_VERSION")`, which pulls in the version from Cargo.toml at build time.

Reply:
If it does work, you can pull in other things like the title and description: https://doc.rust-lang.org/cargo/reference/environment-variables.html